bokeh/server/server.py

```diff
@@ -423,10 +423,16 @@ def __init__(self, applications: Mapping[str, Application | ModifyDoc] | Applica
             context.load_cert_chain(certfile=opts.ssl_certfile, keyfile=opts.ssl_keyfile, password=opts.ssl_password)
             http_server_kwargs['ssl_options'] = context
 
-        sockets, self._port = bind_sockets(opts.address, opts.port)
-        self._address = opts.address
+        if opts.unix_socket:
+            sockets = [bind_unix_socket(opts.unix_socket)]
+            self._unix_socket = opts.unix_socket
+            self._address, self._port = None, None
+            extra_websocket_origins = []
```

Comment:
We will need to say something about this. Before this PR the WS origin allow-list is how to ensure connections can only be initiated from origins that you specify. Presumably, with unix sockets any process with the socket address on the same machine will be able to connect. It would be up to the proxy to enforce these restrictions, if desired? Which is fine, but we need to be explicit about these kinds of consequences and differences.

Also, we also need at least a cursory description of when/why this option exists at all. I presume you want it in order to run behind gunicorn or nginx or something? We need to describe a real-life usage scenario in some detail in the narrative docs (https://docs.bokeh.org/en/latest/docs/user_guide/server.html)

Reply:
Sure, agreed. Yes, I need this to be able to run it behind nginx. I will add some explanation in the docs.

Reply:
Added some docs here, let me know if we can improve something here.